### PR TITLE
correct encoding for string types with bitwidth=0

### PR DESCRIPTION
**Pull request by ravi861**

### CODE REVIEW

File: p4runtime_sh/bytes_utils.py [hunk 1]
```diff
@@ -105,6 +109,8 @@ def make_canonical_if_option_set(bytes_):
 
 
 def parse_value(value_str, bitwidth, base=0):
+    if bitwidth == 0:
+        return str_to_bytes(value_str)
```
Comment (antoninbas) [inline p4runtime_sh/bytes_utils.py:113]: In which case is the bitwidth 0 in the P4Info?

**Review by antoninbas — COMMENTED** [review]
Could you add a unit test for that case?

File: p4runtime_sh/bytes_utils.py (hunk 1 above)
Comment (ravi861) [reply p4runtime_sh/bytes_utils.py:113]: When creating new types like this.
@p4runtime_translation("", string)
type bit<VRF_BITWIDTH> vrf_id_t;

The generated P4info does not have the bitwidth specified.
  new_types {
    key: "vrf_id_t"
    value {
      translated_type {
        sdn_string {
        }
      }
    }
  }

**Review by antoninbas — APPROVED** [review]
LGTM, thanks for adding the test!

File: p4runtime_sh/test.py [hunk 2]
```diff
@@ -417,6 +417,36 @@ def test_table_entry_ternary_dont_care(self):
...
+  exact {
+    value: "16"
+  }
+}
+action {
+  action {
+    action_id: 16783703
+    params {
+      param_id: 1
+      value: "\\x00\\x00\\x11\\x00\\x22\\x33"
```
Comment (antoninbas) [inline p4runtime_sh/test.py:439]: looks like this is wrong

Comment (ravi861) [reply p4runtime_sh/test.py]: Yeah. Just noticed and fixed it. Thanks.

**Pull request merged**